### PR TITLE
solana-contrib: Add support for solscan link generation for `TransactionReceipt`

### PR DESCRIPTION
**Pull request by haodev007**

### CODE REVIEW

**Comment by michaelhly** [comment]
Thanks for the PR. Would you mind fixing the CI failures?

**Comment by haodev007** [comment]
> Thanks for the PR. Would you mind fixing the CI failures?

Can you plz review my updates again? Thanks!

File: packages/solana-contrib/src/transaction/TransactionReceipt.ts [hunk 1]
```diff
@@ -96,7 +101,18 @@ export class TransactionReceipt {
...
+    cluster: Cluster = "mainnet-beta",
+    explorerType: string = ExplorerType.SOLANA_EXPLORER
+  ): string {
+    switch (explorerType) {
+      case ExplorerType.SOLANA_EXPLORER:
+        return `https://explorer.solana.com/tx/${this.signature}?cluster=${cluster}`;
+      case ExplorerType.SOLSCAN:
+        return `https://solscan.io/tx/${this.signature}?cluster=${cluster}`;
+      default:
+        return "";
```
Comment (michaelhly) [inline packages/solana-contrib/src/transaction/TransactionReceipt.ts:115]: Please throw an Error instead.

**Review by michaelhly — CHANGES_REQUESTED** [review]

File: packages/solana-contrib/src/transaction/TransactionReceipt.ts (hunk 1 above)
Comment (haodev007) [reply packages/solana-contrib/src/transaction/TransactionReceipt.ts]: Thanks for your feedback, done it!

File: packages/solana-contrib/src/transaction/TransactionReceipt.ts [hunk 2]
```diff
@@ -96,7 +101,18 @@ export class TransactionReceipt {
...
+    cluster: Cluster = "mainnet-beta",
+    explorerType: string = ExplorerType.SOLANA_EXPLORER
+  ): string {
+    switch (explorerType) {
+      case ExplorerType.SOLANA_EXPLORER:
+        return `https://explorer.solana.com/tx/${this.signature}?cluster=${cluster}`;
+      case ExplorerType.SOLSCAN:
+        return `https://solscan.io/tx/${this.signature}?cluster=${cluster}`;
+      default:
+        throw new Error(`Explorer type is not passed.`);
```
Comment (michaelhly) [inline packages/solana-contrib/src/transaction/TransactionReceipt.ts:115]:
```suggestion
        throw new Error(`Explorer type ${explorerType} is not supported.`);
```

File: packages/solana-contrib/src/transaction/TransactionReceipt.ts [hunk 3]
```diff
@@ -96,7 +101,18 @@ export class TransactionReceipt {
    * @param network
    * @returns
    */
-  generateSolanaExplorerLink(cluster: Cluster = "mainnet-beta"): string {
-    return `https://explorer.solana.com/tx/${this.signature}?cluster=${cluster}`;
```
Comment (michaelhly) [inline packages/solana-contrib/src/transaction/TransactionReceipt.ts:100]: Please put this method back:

```ts
generateSolanaExplorerLink(cluster: Cluster = "mainnet-beta"): string {
    return  generateTXLink(cluster);
```

**Review by michaelhly — CHANGES_REQUESTED** [review]

File: packages/solana-contrib/src/transaction/TransactionReceipt.ts (hunk 3 above)
Comment (haodev007) [reply packages/solana-contrib/src/transaction/TransactionReceipt.ts:100]: Yes, surely

Comment (haodev007) [reply packages/solana-contrib/src/transaction/TransactionReceipt.ts:100]: put that method again there.

File: packages/solana-contrib/src/transaction/TransactionReceipt.ts [hunk 4]
```diff
@@ -96,7 +101,22 @@ export class TransactionReceipt {
    * @param network
    * @returns
    */
+
```
Comment (michaelhly) [inline packages/solana-contrib/src/transaction/TransactionReceipt.ts:104]:
```suggestion
```

**Pull request merged**